src/shared/modules/dbMeta/dbMetaEpics.ts

```diff
@@ -400,23 +410,25 @@ export const dbCountEpic = (some$: any, store: any) =>
     .throttle(() => some$.ofType(DB_META_COUNT_DONE))
     .mergeMap(() =>
       Rx.Observable.fromPromise<void>(
-        new Promise(async resolve => {
+        (async () => {
           store.dispatch(updateCountAutomaticRefresh({ loading: true }))
-          if (getCountAutomaticRefreshEnabled(store.getState())) {
-            const startTime = performance.now()
-            await getNodeAndRelationshipCounts(store)
-            const timeTaken = performance.now() - startTime
 
-            if (timeTaken > 2000) {
-              store.dispatch(updateCountAutomaticRefresh({ enabled: false }))
-            }
-          } else {
-            await getNodeAndRelationshipCounts(store)
+          const res = await getNodeAndRelationshipCounts(store)
+
+          const notAlreadyDisabled = getCountAutomaticRefreshEnabled(
+            store.getState()
+          )
+
+          if (
+            res.requestSucceeded &&
+            res.timeTaken > 1000 &&
```

Comment:
I'm not sure whether this threshold makes sens, I think we could lower it even more actually. In my testing I land somewhere between 10-50m and it should be quick regardless of network and database size, since we only measure computing time and the operation should be a table lookup it